File: DeviceCare/app/src/main/java/com/labs/fi141/devicecare/db/Table.java
package com.labs.fi141.devicecare.db;

import java.util.HashMap;

/**
 * Created by eugenius on 3/19/17.
 */

public class Table {
    public String name;
    public HashMap<String, String> fields;

    public String getCREATE() {
        String str = String.format("create table %s (", name);
        for (String key : fields.keySet()) {
            str += String.format(" %s %s,", key, fields.get(key));
        }
        str = str.substring(0, str.length() - 1);
        str += " );";
        return str;
    }

    public String getDROP() {
        return String.format("drop table if exists %s;", name);
    }

    public String getSELECT() {
        return String.format("select * from %s;", name);
    }

    public String getSELECT(String paramName) {
        return String.format("select * from %s where %s == ?;", name, paramName);
    }

    public String getName() {
        return name;
    }
}

File: DeviceCare/app/src/main/java/com/labs/fi141/devicecare/db/Table.kt
package com.labs.fi141.devicecare.db

import java.util.HashMap

/**
 * Created by eugenius on 3/19/17.
 */

class Table {
    var name: String? = null
    var fields: HashMap<String, String>? = null

    val CREATE: String
        get() {
            var str = "create table ${name} ("
            for (key in fields!!.keys) {
                str += "${key} ${fields!![key]},"
            }
            str = str.subSequence(0, str.length - 1).toString()
            str += ");"
            return str
        }

    val DROP: String
        get() = "drop table if exists ${name};"

    val SELECT: String
        get() = "select * from ${name}"

    fun getSELECT(paramName: String): String {
        return "select * from ${name} where ${paramName} == ?"
    }
}
